mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -63,62 +69,71 @@ public class ExecutorStateManagerImpl implements ExecutorStateManager {
         .build();
 
     @Override
-    public void putIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state) {
+    public void trackIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state) {
         this.taskExecutorStateMap.putIfAbsent(taskExecutorID, state);
         if (this.archivedState.getIfPresent(taskExecutorID) != null) {
             log.info("Reviving archived executor: {}", taskExecutorID);
             this.archivedState.invalidate(taskExecutorID);
         }
 
-        // add to buckets. however new executors won't have valid registration at this moment and requires marking
-        // again later when registration is ready.
+        tryMarkAvailable(taskExecutorID, state);
+    }
+
+    /**
+     * Add to buckets. however new executors won't have valid registration at this moment and requires marking
+     * again later when registration is ready.
+     * @param taskExecutorID taskExecutorID
+     * @param state state
+     * @return whether the target executor is marked as available.
+     */
+    public boolean tryMarkAvailable(TaskExecutorID taskExecutorID, TaskExecutorState state) {
         if (state.isAvailable() && state.getRegistration() != null) {
```

Comment:
Semantically, it does not make sense that we are calling this method to make the task executor available, and the state could in fact be non-available. I would probably do the checks from the caller.

Reply:
let me switch this to private. Should be used inside the TaskExecutorImpl only.